src/directives/column-reorder.js

```diff
@@ -0,0 +1,223 @@
+                        var rowTemplate = templateMapper[rowId];
+                        rowTemplate = angular.element('<tr>' + rowTemplate + '<tr>');
+                        rowTemplate = angular.element(rowTemplate[0].children);
+
+                        var columnCount = rowTemplate.length;
+                        if (oldColIndex >= columnCount || newColIndex >= columnCount) {
+                            return;
+                        }
+
+                        if (offset < 0) {
+                            // Insert Before
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex, 0, oldColumn);
+                        } else if (offset > 0) {
+                            // Insert After
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex + 1, 0, oldColumn);
+                        } else {
+                            // Swap Column Places
+                            var oldColumn = rowTemplate.splice(oldColIndex, 1)[0];
+                            var newColumn = rowTemplate.splice(newColIndex, 1)[0];
+                            rowTemplate.splice(newColIndex, 0, oldColumn);
+                            rowTemplate.splice(oldColIndex, 0, newColumn);
+                        }
+
+                        var newRowTemplate = '';
+                        for(var i = 0; i < rowTemplate.length; i++) {
+                            newRowTemplate += rowTemplate[i].outerHTML;
+                        }
+
+                        templateMapper[rowId] = newRowTemplate;
+                    });
+
+                    $scope.$broadcast('drawTableRows');
+                }
+
+            },
+            compile: function($element, $attrs) {
+                var headerRow = $element[0].querySelector('thead tr');
+                headerRow.setAttribute('reorderable-column-template-mapper', getId());
+
+                var headers = headerRow.querySelectorAll('th');
+                var headerCount = headers.length;
+                while (headerCount--) {
+                    headers[headerCount].setAttribute('reorderable-header-drag-drop', '');
+                }
+
+                var bodyRows = $element[0].querySelectorAll('tbody tr');
+                var bodyRowCount = bodyRows.length;
+                while (bodyRowCount--) {
+                    bodyRows[bodyRowCount].setAttribute('reorderable-column-template-mapper', getId());
+                }
+
+            }
+        }
+    });
+
+    module.directive('reorderableColumnTemplateMapper', function($compile) {
+        return {
+            require: '^reorderableColumns',
+            compile: function($element, $attrs) {
+                var directiveName = this.name;
+                var rowId = $attrs[directiveName];
+
+                var uncompiledHtml = $element[0].innerHTML;
```

Comment:
so I think we push/merge can you look at how we can modify the hidden columns to support this?  I take it the issue here is the uncompiledHtml is static with the hidden column (or visible) and if you change its status it is not reflected in the uncompiledHtml?  We should not merge/push without the two working properly together.